app/services/action_handlers/pull_request.rb

```diff
@@ -0,0 +1,46 @@
+module ActionHandlers
+  class PullRequest < ActionHandler
+    ACTIONS = %w[open review_requested closed \
+                 review_request_removed].freeze
+    private_constant :ACTIONS
+
+    def resolve
+      return unless handleable?
+
+      handle_action
+    end
+
+    private
+
+    def handleable?
+      ACTIONS.include?(@payload['action'])
+    end
+
+    def open
+      @event_type.open!
+      @event_type.update!(opened_at: Time.current)
+    end
+
+    def merged
+      @event_type.update!(merged_at: Time.current)
+    end
+
+    def closed
+      merged if @payload['pull_request']['merged'] == true
+      @event_type.closed!
+      @event_type.update!(closed_at: Time.current)
+    end
+
+    def review_request_removed
+      reviewer = User.find_by!(github_id: @payload['requested_reviewer']['id'])
+      @event_type.review_requests.find_by!(reviewer: reviewer).removed!
+    end
+
+    def review_requested
+      owner = find_or_create_user(@payload['pull_request']['user'])
+      reviewer = find_or_create_user(@payload['requested_reviewer'])
+      @event_type.review_requests.create!(owner: owner, reviewer: reviewer,
+                                          node_id: reviewer.node_id, login: reviewer.login)
```

Comment:
I believe it is the main purpose of the method, to create a `review_request`, and is in the only part of the app where is being used.

Reply:
Yes, the concern is related to the ownership of the `review_request` collection and in a more abstract sense with the question 'which object should modify the contents of a given collection (persisted or not)?'.

If the object PullRequest has the ownership of the `review_requests` collection it is an interesting idea that the only object modifying the collection be the PullRequest itself.
For that it would have a method call `create_review_request`.

If the object PullRequest does not have the ownership of that collection then the single point can be in the collection owner(s).